mantidproject/mantid
# Improvements to per angle options

This PR adds the following changes to the ISIS Reflectometry GUI:

- Add additional options that can be set on a per-angle basis on the Settings tab: Q min, Q max, dQ/Q and Scale.
- Fix a bug where the Get Defaults button on the Settings tab clears values for which a default doesn't exist
- If duplicate angles are given, the first one takes precedence

The large number of files changed is due to renaming a pass-through funtion.

**To test:**
Test default options:
- Open the `ISIS Reflectometry` GUI
- Set the instrument to `INTER` and enter a Run number `13460`
- On the Settings tab, in the table enter some values, e.g.
  - Transmission runs = `13463,13464`
  - Q min = 0.008
  - Q max = 0.1
  - dQ/Q = 0.02
  - Scale = 0.9
- On the Runs tab, click Process
- Check that all the values are populated with the same values in the table on the Runs tab (apart from transmission runs - historically this was never populated)
- Check that a workspace `TRANS_13463+13464` has been created. This indicates that the correct transmission runs were used.

Test per-angle lookup:
- Delete all workspaces, and clear all values in the Runs tab table apart from the run number and the angle. 
- On the Settings tab, leave the values as before but add `Angle = 0.7`
- Add another row with the same angle and different values.
- Add another row with a different angle and different values.
- Re-process and check you get the same results. This checks that the angle lookup found the correct row and ignored the second duplicate angle.

Test defaults:
- On the settings tab, click Get Defaults. The values in the table should remain unchanged because they do not have values in the INTER parameters file.

Fixes #22039 

**Release Notes** 
*Updated release notes*

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
